lua/neogit/lib/finder.lua

```diff
@@ -0,0 +1,178 @@
+local config = require("neogit.config")
+local async = require("plenary.async")
+
+local function telescope_mappings(on_select, allow_multi)
+  local action_state = require("telescope.actions.state")
+  local actions = require("telescope.actions")
+  --- Lift the picker select action to a item select action
+  local function select_action(prompt_bufnr)
```

Comment:
I moved this from the `fuzzy_finder.lua` since it is only needed for telescope, the `vim.ui.select` calls the select function with the item directly